### ISSUE TITLE
Support custom docker images

### ISSUE THREAD

**Issue by felipecrs** [gitpod-io/gitpod#3353]
This issue is to track whatever is needed to support custom docker images to be used in Gitpod. It could since a simple guide on what's need for the custom image, or actually a fix in Gitpod side to support it.

For example, I have a docker image ([ghcr.io/felipecrs/devcontainer](https://github.com/users/felipecrs/packages/container/package/devcontainer)) which I use as base for most of my projects with VSCode Devcontainers (Codespaces + Remote Containers). I would like to use the same image in Gitpod, as it has many of the things that I need.

For example, if Gitpod expects an user called `gitpod`, it could be changed to expect an user with UID of `1000` instead (my image uses `vscode` as the non-root user name).

**Comment by davidwindell** [comment]
I agree this would be great to document, here's the key changes I had to make https://github.com/outeredge/edge-docker-php-dev/commit/af603b07522d275897cb3e88aba79936e4de649f#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R23 to our dev image. But it took a while to find the layer.sh info and discover I would have to create a user with id `33333` for everything to work as I expected.

**Comment by geropl** [comment]
> It could since a simple guide on what needs to be set in the custom image

Thx @felipecrs for reporting. IMO better documentation is key here, as there are only so much image configurations we can handle. E.g., we already have a special case for `root` being the only user, but to for `1000` as you suggested.
I renamed the title and added to your roadmap.
